micrometer-meter-provider/src/main/java/io/opentelemetry/contrib/metrics/micrometer/internal/instruments/MicrometerDoubleGauge.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.metrics.micrometer.internal.instruments;
+
+import io.micrometer.core.instrument.MultiGauge;
+import io.micrometer.core.instrument.Tags;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleGaugeBuilder;
+import io.opentelemetry.api.metrics.LongGaugeBuilder;
+import io.opentelemetry.api.metrics.ObservableDoubleGauge;
+import io.opentelemetry.api.metrics.ObservableDoubleMeasurement;
+import io.opentelemetry.api.metrics.ObservableLongGauge;
+import io.opentelemetry.api.metrics.ObservableLongMeasurement;
+import io.opentelemetry.contrib.metrics.micrometer.internal.state.InstrumentState;
+import io.opentelemetry.contrib.metrics.micrometer.internal.state.MeterSharedState;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+public final class MicrometerDoubleGauge extends AbstractInstrument {
+  private final MultiGauge multiGauge;
```

Comment:
Just wondering: wouldn't implementing the gauges in the same way as all other instruments be simpler? Why introduce a `MultiGauge` here?

Reply:
MultiGauge was suggested by Micrometer, I don't mind revisiting this though for the sake of simplicity.